include/ccf/crypto/jwk.h

```diff
@@ -27,13 +27,27 @@ namespace ccf::crypto
     JsonWebKeyType kty;
     std::optional<std::string> kid = std::nullopt;
     std::optional<std::vector<std::string>> x5c = std::nullopt;
-    std::optional<std::string> issuer = std::nullopt;
 
     bool operator==(const JsonWebKey&) const = default;
   };
   DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(JsonWebKey);
   DECLARE_JSON_REQUIRED_FIELDS(JsonWebKey, kty);
-  DECLARE_JSON_OPTIONAL_FIELDS(JsonWebKey, kid, x5c, issuer);
+  DECLARE_JSON_OPTIONAL_FIELDS(JsonWebKey, kid, x5c);
+
+  struct JsonWebKeyExtended
```

Comment:
That seems to duplicate JsonWebKeyRSAPublic and create another hierarchy, it would be nice if we could unify our handling of JWK serdes, or at least avoid fragmenting it further. It's not a requirement for this PR, but it's worth spending a little time on I suspect.

Reply:
Also this looks like it means we wouldn't handle EC or Ed keys in OpenID.

Reply:
Replied/solved in the comment above https://github.com/microsoft/CCF/pull/6680#discussion_r1897876282